lib/improver/nbhood/vicinity.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Module containing to apply neighbourhood processing to a vicinity."""
+
+import iris
+
+from improver.utilities.spatial import OccurrenceWithinVicinity
+from improver.nbhood.nbhood import NeighbourhoodProcessing
+
+
+class ProbabilityOfOccurrence(object):
+
+    """
+    Calculate the probability of a phenomenon occurring within specified
+    distance.
+    """
+
+    def __init__(self, distance, neighbourhood_method, radii, lead_times=None,
+                 unweighted_mode=False, ens_factor=1.0):
+        """
+        Initialise the class.
+
+        Args:
+            distance : float
+                Distance in metres used to define whether there is an
+                occurrence within the vicinity.
+            neighbourhood_method : str
+                Name of the neighbourhood method to use. Options: 'circular',
+                'square'.
```

Comment:
At the moment, I've left it so that it is possible to get the probability of occurrence within a square vicinity whilst then applying a circular neighbourhood.